src/plugins/borrow-plugins/types.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import { type EdgeAccount, type EdgeCurrencyWallet, type EdgeNetworkFee } from 'edge-core-js'
+
+// Proposal for a "service plugin" which is a Higher-Order-Plugin that can be
+// instantiated with some account context and returns a service instance.
+export type EdgeServicePlugin<Info, Api> = {
+  info: Info,
+  init: (account: EdgeAccount) => Promise<Api>
+}
+
+// Borrow more:
+export type BorrowRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Make payment:
+export type RepayRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Deposit collateral:
+export type DepositRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional wallet which will deposit on behalf of the loan account mainWallet
+  fromWallet?: string
+}
+
+// Withdraw collateral:
+export type WithdrawRequest = {
+  tokenId?: string,
+  nativeAmount: string,
+
+  // Optional destination for the funds
+  toWallet?: string
+}
+
+// General purpose approvable action
+export type ApprovableAction = {
+  +networkFee: EdgeNetworkFee,
+  +approve: () => Promise<void>
+}
+
+//
+// Loan Account – Primary entity
+//
+
+export type LoanAccount = {
+  // Account wallet for the loan (e.g. Ethereum wallet)
+  mainWallet: EdgeCurrencyWallet,
+
+  collaterals: Array<{
+    tokenId?: string,
+    nativeAmount: string
+  }>,
+  debts: Array<{
+    tokenId?: string,
+    nativeAmount: string,
+    apy: number
+  }>,
+
+  loanToValue: number,
+
+  getApr: (tokenId?: string) => Promise<number>,
```

Comment:
I am not sure what this does. Does this provide an estimate for new debt? Otherwise, we have `apy` in the `debts` array already. Also, why is one compounding and the other not?

Reply:
A user may not have any `debts` and the GUI will want the APR/APY for a particular token

Reply:
Change to `getAprQuote` and added a comment